### PR TITLE
Improve REPL completions

### LINKED ISSUE #216: BoundsError: attempt to access 2-element Array
## Details
- Atom version: 1.33.1
- Julia version: 1.0.1
- OS: Mac OS
- Package versions:
  - Atom.jl: v0.7.11
  - julia-client: 0.7.11
  - ink: 0.9.13

### Steps to reproduce

1.  run the following

```julia
ex = :(function foo(::Int) end)
```

2. type the following

```
ex.args[1].args[2].
```

### PR DESCRIPTION
**Pull request by pfitzseb**
Fixes https://github.com/JunoLab/Juno.jl/issues/216. Might also help with https://github.com/JuliaLang/julia/issues/28931.

~This probably isn't the *right* way to fix this, but I'm not sure where the error originates either...~

### CODE REVIEW

File: stdlib/REPL/src/REPLCompletions.jl [hunk 1]
```diff
@@ -432,7 +432,14 @@ function complete_methods(ex_org::Expr, context_module=Main)::Vector{Completion}
     args_ex = Any[]
     func, found = get_value(ex_org.args[1], context_module)
     !found && return Completion[]
-    for ex in ex_org.args[2:end]
+
+    funargs = ex_org.args[2:end]
+    # handle broadcasting
+    if ex_org.head === :. && ex_org.args[2] isa Expr
+        funargs = ex_org.args[2].args
```
Comment (JeffBezanson) [inline stdlib/REPL/src/REPLCompletions.jl]: This doesn't seem right to me --- in `f.(args` the function won't necessarily be called on those arguments, but their elements (if arrays). This is a bit tricky to handle, so I'd leave it out for now.

Comment (pfitzseb) [reply stdlib/REPL/src/REPLCompletions.jl]: Doh, of course. I've pushed a commit such that we always return *all* methods that match the *number* of arguments and completely disregard any argument types.

IMHO that's better than not having any autocompletions for broadcasted functions, but lemme know if you don't think so and I'll remove that codepath :)

**Comment by pfitzseb** [comment]
Bump.

**Comment by vtjnash** [comment]
Related to https://github.com/JuliaLang/julia/issues/24257 also?

**Review by vtjnash — APPROVED** [review]
Don't see anything objectionable here—it should at least be an improvement.

**Pull request merged**